libs/common/RespReadUtils.cs

```diff
@@ -212,15 +211,46 @@ private static bool TryReadInt(ref byte* ptr, byte* end, out int value, out ulon
         /// <summary>
         /// Tries to read a RESP length header from the given ASCII-encoded RESP string
         /// and, if successful, moves the given ptr to the end of the length header.
+        /// NOTE: It will throws an exception if length header is negative.
+        /// </summary>
+        /// <param name="length">If parsing was successful, contains the extracted length from the header.</param>
+        /// <param name="ptr">The starting position in the RESP string. Will be advanced if parsing is successful.</param>
+        /// <param name="end">The current end of the RESP string.</param>
+        /// <param name="isArray">Whether to parse an array length header ('*...\r\n') or a string length header ('$...\r\n').</param>
+        /// <returns>True if a length header was successfully read.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool ReadUnsignedLengthHeader(out int length, ref byte* ptr, byte* end, bool isArray = false)
+        {
+            length = -1;
+            if (ptr + 3 > end)
+                return false;
+
+            var readHead = ptr + 1;
+            var negative = *readHead == '-';
+
+            if (!ReadSignedLengthHeader(out length, ref ptr, end, isArray))
+                return false;
+
+            if (negative)
```

Comment:
You can put this before the ReadSignedLengthHeader call

Reply:
Good point, I just pushed a fix.